src/main/java/org/eclipse/golo/runtime/RegularMethodFinder.java

```diff
@@ -113,69 +88,21 @@ private static String reverseStructAugmentation(String receiverClassName) {
       + "$" + receiverClassName.replace('.', '$');
   }
 
-  private List<Method> getCandidates() {
-    List<Method> candidates = new LinkedList<>();
-    Set<Method> methods = new HashSet<>();
-    Collections.addAll(methods, receiverClass.getMethods());
-    Collections.addAll(methods, receiverClass.getDeclaredMethods());
-    for (Method method : methods) {
-      if (isCandidateMethod(method) && !method.isVarArgs()) {
-        candidates.add(method);
-      } else if (isValidPrivateStructAccess(method) && !method.isVarArgs()) {
-        candidates.add(method);
-      }
-    }
-    for (Method method : methods) {
-      if (isCandidateMethod(method) && method.isVarArgs()) {
-        candidates.add(method);
-      } else if (isValidPrivateStructAccess(method) && method.isVarArgs()) {
-        candidates.add(method);
-      }
-    }
-    return candidates;
-  }
 
-  private MethodHandle findInMethods() throws IllegalAccessException {
-    List<Method> candidates = getCandidates();
-    if (candidates.isEmpty()) { return null; }
-    if (candidates.size() == 1) { return toMethodHandle(candidates.get(0)); }
-
-    for (Method method : candidates) {
-      if (isMethodDecorated(method)) {
-        return toMethodHandle(method);
-      }
-      Class<?>[] parameterTypes = method.getParameterTypes();
-      Object[] argsWithoutReceiver = copyOfRange(args, 1, args.length);
-      if (TypeMatching.argumentsNumberMatch(argsWithoutReceiver, method, parameterTypes)) {
-        if (TypeMatching.canAssign(parameterTypes, argsWithoutReceiver, method.isVarArgs())) {
-          return toMethodHandle(method);
-        }
-      }
-    }
-    return null;
+  private Stream<Optional<MethodHandle>> findInMethods() {
```

Comment:
I propose to return `Method` instead of `MethodHandle` see [my comment](https://github.com/eclipse/golo-lang/pull/332#issuecomment-166919207) on #332.